### PR TITLE
Added aspect ratio preserving option for web_file_size.

### PR DESCRIPTION
**Pull request by CeltonMcGrath**

### CODE REVIEW

File: src/pyasm/biz/file.py [hunk 1]
```diff
@@ -619,9 +619,17 @@ def _process_pdf(my, file_name):
                 import shlex, subprocess
                 subprocess.call([convert_exe, '-geometry','80','-raise','2x2','%s[0]'%my.file_path,\
                         "%s"%tmp_icon_path]) 
+                
+                # Shrink image based of web_file_size
```
Comment (CeltonMcGrath) [inline src/pyasm/biz/file.py]: typo

File: src/pyasm/biz/file.py [hunk 2]
```diff
@@ -644,16 +652,17 @@ def get_web_file_size(my):
...
             if len(parts) == 2:
                 try:
                     thumb_size = (int(parts[0]), int(parts[1]))
                 except ValueError:
                     thumb_size = (640, 480)
 
-        return thumb_size
-
+        return thumb_size 
+
```
Comment (CeltonMcGrath) [inline src/pyasm/biz/file.py]: remove white space

File: src/pyasm/biz/file.py [hunk 3]
```diff
@@ -674,11 +683,70 @@ def _process_video(my, file_name):
...
+                elif not free_aspect_ratio and (width > max_width or height > max_height):
+                    size_option = "-s"
+                    size =  "%sx%s" % (thumb_web_size[0], thumb_web_size[1])
+                else: 
+                    if width > height:
+                        size_option = "-vf"
+                        size = "scale=%s:-1" % thumb_web_size[0]
+                        
+                    elif height > width:
+
```
Comment (CeltonMcGrath) [inline src/pyasm/biz/file.py]: remove line

File: src/pyasm/biz/file.py [hunk 4]
```diff
@@ -674,11 +683,70 @@ def _process_video(my, file_name):
...
+                streams = data.get("streams") or []
+                sample_stream = streams[0]
+                width = int(sample_stream.get("width"))
+                height = int(sample_stream.get("height"))
             
-           
+                max_width = thumb_web_size[0]
+                max_height = max_width*1 if free_aspect_ratio else thumb_web_size[1]
+               
+                if width < max_width and height < max_height:
```
Comment (CeltonMcGrath) [inline src/pyasm/biz/file.py]: Add comment for cases

File: src/pyasm/biz/file.py [hunk 5]
```diff
@@ -619,9 +619,17 @@ def _process_pdf(my, file_name):
...
                         "%s"%tmp_icon_path]) 
+                
+                # Shrink image based on web_file_size
+                # (preserves aspect ratio regardless)
+                pdf_width = thumb_web_size[0]
+                if thumb_web_size[1] == -1:
+                    pdf_height = pdf_width*10
+                else:
+                    pdf_height = thumb_web_size[1]
+                size = '%sx%s>' % (pdf_width, pdf_height)
```
Comment (oboreo) [inline src/pyasm/biz/file.py]: why does this size have an extra > ?

**Comment by oboreo** [comment]
This is ideally done on branch 4.6 instead

**Pull request merged**